portal-impl/src/com/liferay/portlet/internal/PortletRequestImpl.java

```diff
@@ -800,10 +858,129 @@ else if (themeDisplay.isLifecycleAction()) {
 		_locale = themeDisplay.getLocale();
 		_plid = plid;
 
-		// TODO Set render parameter values
+		Set<String> publicRenderParameterNames = new HashSet<>();
```

Comment:
@shuyangzhou FYI -- in a future PR, PortletURLImpl it calls `_portletRequest.getRenderParameters()` in order to take advantage of all the render parameter processing work that is done here in `PortletRequestImpl`. This is a nice optimization because it enables us to do all the render parameter processing work once.